source/Octopus.Tentacle/Services/ServicesModule.cs

```diff
@@ -15,11 +17,10 @@ protected override void Load(ContainerBuilder builder)
 
             builder.RegisterType<NuGetPackageInstaller>().As<IPackageInstaller>();
 
-            // Find anything with the [Service] attribute
-            builder.RegisterAssemblyTypes(ThisAssembly)
-                .Where(t => t.GetCustomAttributes(typeof (ServiceAttribute), true).Length > 0)
-                .Named(t => "I" + t.Name, typeof (object))
-                .SingleInstance();
+            // Register our Halibut services
+            var serviceTypes = ThisAssembly.GetTypes().Where(t => t.GetCustomAttributes(typeof(ServiceAttribute), true).Length > 0).ToArray();
+            var assemblyServices = new KnownServiceSource(serviceTypes);
+            builder.RegisterInstance(assemblyServices).AsImplementedInterfaces();
```

Comment:
These don't need to be single instances anymore?

Reply:
Pretty sure RegisterInstance takes care of that :)

Reply:
lol yes 😅